### PR TITLE
DE43824 - Retract with unsaved changes dialogue

### PR DESCRIPTION
**Pull request by tdevos-d2l**
[DE43824](https://rally1.rallydev.com/#/?detail=/defect/600942267527&fdp=true)

User is now prompted with a dialogue when retracting with unsaved changes as shown in the image below.

image

General flow for detecting unsaved changes and displaying the dialog was followed closely to when a user tries to navigate with unsaved changes.
Wording in the dialog was approved by Megan Shuker.

### CODE REVIEW

**Comment by tdevos-d2l** [comment]
Not sure if a new unit test will be needed for these changes. Let me know if there is a new use case that would benefit from having one and I'll add it.

File: lang/en.js [hunk 1]
```diff
@@ -156,9 +157,12 @@ export default {
 	"unflaggedFile": "{fileName} unflagged",
 	"unsavedAnnotationsBody": "Are you sure you want to leave without saving your annotation changes?",
 	"unsavedAnnotationsDiscardButton": "Discard annotation changes",
+	"unsavedChangesDiscardAndRetractButton": "Discard and Retract",
+	"unsavedChangesDiscardButton": "Discard changes",
 	"unsavedAnnotationsTitle": "Unsaved annotation changes",
```
Comment (maddiegabriel) [inline lang/en.js:162]: Should we update this dialog to use title case too while we're at it? Along with `unsavedAnnotationsDiscardButton`

**Review by maddiegabriel — APPROVED** [review]
LGTM! 👍

**Review by JamieTran — APPROVED** [review]
Great work!!

**Pull request merged**